File: app/src/main/java/br/com/star_wars_wiki/database/dao/PeopleDAO.java
package br.com.star_wars_wiki.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import br.com.star_wars_wiki.entity.People;

@Dao
public interface PeopleDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllPeople(List<People> list);

    @Query("SELECT * FROM People")
    LiveData<List<People>> getAllPeople();
}

File: app/src/main/java/br/com/example/starwars/data/local/PeopleDao.kt
package br.com.example.starwars.data.local

import androidx.paging.PagingSource
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import br.com.example.starwars.data.entities.ApiPeople

@Dao
interface PeopleDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(apiPeople: List<ApiPeople>)

    @Query("SELECT * FROM people")
    fun getListPeople(): PagingSource<Int, ApiPeople>

    @Query("UPDATE people SET favorite = :favorite WHERE id = :id")
    suspend fun updatePerson(favorite: Boolean, id: Int)
}